### DISCUSSION TITLE
Installation help

### DISCUSSION THREAD

**Discussion by GideonBear** [ouch-org/ouch#366] [Q&A]
`cargo install ouch` fails with the same error as #256. Which is logical, since I'm on an old Debian version with an old `zstd` version.
The `musl` package works, but how do I install it? There are no instructions or an installation script. Just moving the `ouch` executable into a `bin` works, but that doesn't install the `completions` and `man`.
Thanks for the great tool!

**Comment by figsoda** [comment]
Just FYI that `cargo install ouch` also doesn't install completions or man pages. We currently don't have an installation script and are relying on downstream maintainers for completions and man pages. I'm not familiar with debian, but manually copying the completion files and man pages to the directories should work in the meantime. The completions might also work if you source the `ouch.bash` file in you bashrc.

**Comment by marcospb19** [comment]
We [used to have an installation script](https://github.com/ouch-org/ouch/pull/310), but we got rid of it in favor of just having proper packages, here's a list of problems that I found with my script:

1. Cumbersome to maintain, and we don't have many maintainers.
2. Shell completions and man pages directories are different between systems, which makes things a lot harder (and impossible for some systems like `NixOS`).
3. Installing requires `sudo`, which is kinda sketchy to use in an installation script.
4. We don't have a good way of testing the script in multiple systems.

Between mainstream distributions, I believe Debian is the hardest one for us to ship Ouch to, and even if we did, it would be out of date.

From the README, these are the packages people made for Ouch (at the time of this writing): 

![image](https://user-images.githubusercontent.com/38900226/219884705-8dc25e63-405a-4c0b-9180-a9b0040c18ec.png)

No Debian, so we don't have a good solution for this yet. Perhaps in README we could link to a wiki page giving instructions to people on what to do in Debian/Ubuntu if they also want to manually install completions and man pages (but I don't have either installed here).

**Reply by GideonBear** [reply]
Maybe `zstd` should be noted in the [dependencies section of the README](https://github.com/ouch-org/ouch#dependencies), in combination with a minimal version, so people don't have to search the issues to find #256. This would make it a lot easier to build it from source using `cargo install`.

**Reply by figsoda** [reply]
The dependencies section actually seems outdated. I don't have any of the libraries installed on my system (not even pkg-config), but `cargo install ouch` works as expected. The issue seems to not be missing dependencies, and something went wrong in the linking process.
